src/agreement/verifiers.rs

```diff
@@ -92,53 +85,47 @@ async fn verify_step_votes(
         Ok(sub_committee)
     }?;
 
-    unsafe {
-        // aggregate public keys
-        let apk = aggregate_pks(committees_set.clone(), sub_committee).await?;
+    // aggregate public keys
+    let apk = aggregate_pks(committees_set.clone(), sub_committee).await?;
 
-        // verify signatures
-        if let Err(e) = verify_signatures(hdr.round, step, hdr.block_hash, apk, sv.signature) {
-            error!("verify signatures fails with err: {}", e);
-            return Err(Error::VerificationFailed);
-        }
+    // verify signatures
+    if let Err(e) = verify_signatures(hdr.round, step, hdr.block_hash, apk, sv.signature) {
+        error!("verify signatures fails with err: {}", e);
+        return Err(Error::VerificationFailed);
     }
 
     // Verification done
     Ok(())
 }
 
-async unsafe fn aggregate_pks(
+async fn aggregate_pks(
     committees_set: Arc<Mutex<CommitteeSet>>,
-    subcomittee: Cluster<PublicKey>,
+    subcomittee: Cluster<ConsensusPublicKey>,
 ) -> Result<dusk_bls12_381_sign::APK, Error> {
     let pks = {
         let mut pks = vec![];
 
         let guard = committees_set.lock().await;
         let provisioners = guard.get_provisioners();
 
-        for (pubkey, _) in subcomittee.into_iter() {
-            let m = provisioners
-                .get_member(&pubkey)
-                .expect("raw public key not found");
-            pks.push(dusk_bls12_381_sign::PublicKey::from_slice_unchecked(
-                &m.get_raw_key(),
-            ));
+        for (pubkey, _) in subcomittee.iter() {
+            provisioners
+                .get_member(pubkey)
+                .ok_or(Error::NotCommitteeMember)?;
```

Comment:
If we don't need to map pubkey to raw_pubkey, provisioners.get_member(..) is redundant here.

Reply:
Removed the get_member and the guard too